src/webgl/loading.js

```diff
@@ -241,6 +265,300 @@ function parseObj(model, lines) {
   return model;
 }
 
+/**
+ * STL files can be of two types, ASCII and Binary,
+ *
+ * We need to convert the arrayBuffer to an array of strings,
+ * to parse it as an ASCII file.
+ */
+function parseSTL(model, buffer) {
+  if (isBinary(buffer)) {
+    parseBinarySTL(model, buffer);
+  } else {
+    var reader = new DataView(buffer);
+    var decoder = new TextDecoder('utf-8');
+    var lines = decoder.decode(reader);
+    var lineArray = lines.split('\n');
+    parseASCIISTL(model, lineArray);
+  }
+  return model;
+}
+
+/**
+ * This function checks if the file is in ASCII format or in Binary format
+ *
+ * It is done by searching keyword `solid` at the start of the file.
+ *
+ * An ASCII STL data must begin with `solid` as the first six bytes.
+ * However, ASCII STLs lacking the SPACE after the `d` are known to be
+ * plentiful. So, check the first 5 bytes for `solid`.
+ *
+ * Several encodings, such as UTF-8, precede the text with up to 5 bytes:
+ * https://en.wikipedia.org/wiki/Byte_order_mark#Byte_order_marks_by_encoding
+ * Search for `solid` to start anywhere after those prefixes.
+ */
+function isBinary(data) {
+  var reader = new DataView(data);
+
+  // US-ASCII ordinal values for `s`, `o`, `l`, `i`, `d`
+  var solid = [115, 111, 108, 105, 100];
+  for (var off = 0; off < 5; off++) {
+    // If "solid" text is matched to the current offset, declare it to be an ASCII STL.
+    if (matchDataViewAt(solid, reader, off)) return false;
+  }
+
+  // Couldn't find "solid" text at the beginning; it is binary STL.
+  return true;
+}
+
+/**
+ * This function matches the `query` at the provided `offset`
+ */
+function matchDataViewAt(query, reader, offset) {
+  // Check if each byte in query matches the corresponding byte from the current offset
+  for (var i = 0, il = query.length; i < il; i++) {
+    if (query[i] !== reader.getUint8(offset + i, false)) return false;
+  }
+
+  return true;
+}
+
+/**
+ * This function parses the Binary STL files.
+ * https://en.wikipedia.org/wiki/STL_%28file_format%29#Binary_STL
+ *
+ * Currently there is no support for the colors provided in STL files.
+ */
+function parseBinarySTL(model, buffer) {
+  var reader = new DataView(buffer);
+
+  // Number of faces is present following the header
+  var faces = reader.getUint32(80, true);
+  var r,
```

Comment:
Per vertex color isn't enabled in GL currently. I think that it might be a nice addition eventually but that seems like its own separate task. It is is probably best to remove all of the color parsing code for now. If iper-vertex-color gets added down the line then this code can get added back in.  If you'd like to archive this work you could push this branch to another before removing the code. Then start an issue here requesting that per-vertex color is added to GL, and add a link to your archive branch for reference. I know this is a bit of work but it would be helpful. Thank you!

Reply:
@aferriss and I were discussing this topic exactly. The thing is, If I don't read the color bits if they are present, they will create an offset and those color bits will be computed as vertices, so I have to atleast read them, and I thought If I am already reading them, why not compute them too, as in the future if the feature is added, It will be useful. Performance wise I don't think they affect too much right now. I will add the comments so that in future it will be easier to activate them.

Reply:
That makes sense!